core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -910,6 +911,23 @@ class CivilizationInfo : IsPartOfGameInfoSerialization {
     fun updateViewableTiles() = transients().updateViewableTiles()
     fun updateDetailedCivResources() = transients().updateCivResources()
 
+    fun doTurn(): Boolean {
+
+        // Defeated civs do nothing
+        if (isDefeated())
+            return false
+
+        // Do stuff
+        NextTurnAutomation.automateCivMoves(this)
+
+        // Update barbarian camps
+        if (isBarbarian() && !gameInfo.gameParameters.noBarbarians)
+            gameInfo.barbarians.updateEncampments()
+
+        // Have we won?
+        return victoryManager.hasWon()
+    }
```

Comment:
"do turn" and "check has won" are two very different things, we shouldn't be putting them into the same function

Reply:
Done